core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -1793,14 +1793,23 @@ object SnappySession extends Logging {
       override def load(key: CachedKey): (CachedDataFrame,
           Map[String, String]) = {
         val session = key.session
+        session.disableStoreOptimizations = false
         val df = session.executeSQL(key.sqlText)
         val plan = df.queryExecution.executedPlan
         // if this has in-memory caching then don't cache the first time
         // since plan can change once caching is done (due to size stats)
         if (plan.find(_.isInstanceOf[InMemoryTableScanExec]).isDefined) {
           (null, null)
         } else {
-          evaluatePlan(df, session, key.sqlText, key)
+          try {
+            evaluatePlan(df, session, key.sqlText, key)
+          } catch {
+            case e: CodeCompileException => {
+              session.disableStoreOptimizations = true
+              val df = session.executeSQL(key.sqlText)
+              evaluatePlan(df, session, key.sqlText, key)
```

Comment:
Any specific reason to not make disableStoreOptimizations as false after evaluatePlan, may be in a try finally block?